<?xml version="1.0" encoding="UTF-8"?><testrun duration="5717" footerText="Generated by IntelliJ IDEA on 25/08/24, 11:34â€¯am" name="ServiceImplDiffblueTest">
    <count name="total" value="51"/>
    <count name="passed" value="51"/>
    <config nameIsGenerated="true" configId="JUnit" name="ServiceImplDiffblueTest">
        <module name="HospitalManagementSystemWebApp"/>
        <extension name="coverage">
            <pattern>
                <option name="PATTERN" value="com.hospital.service.*"/>
                <option name="ENABLED" value="true"/>
            </pattern>
        </extension>
        <option name="PACKAGE_NAME" value="com.hospital.service"/>
        <option name="MAIN_CLASS_NAME" value="com.hospital.service.ServiceImplDiffblueTest"/>
        <option name="TEST_OBJECT" value="class"/>
        <method v="2">
            <option name="Make" enabled="true"/>
        </method>
    </config>
    <root name="ServiceImplDiffblueTest" comment="com.hospital.service" location="java:suite://com.hospital.service.ServiceImplDiffblueTest">
        <output type="stdout">E:\java_17\bin\java.exe -ea -Didea.test.cyclic.buffer.size=1048576 "-javaagent:E:\IntelliJ IDEA 2024.2.0.1\lib\idea_rt.jar=61261:E:\IntelliJ IDEA 2024.2.0.1\bin" -Dfile.encoding=UTF-8 -classpath "C:\Users\User\.m2\repository\org\junit\platform\junit-platform-launcher\1.10.0\junit-platform-launcher-1.10.0.jar;E:\IntelliJ IDEA 2024.2.0.1\lib\idea_rt.jar;E:\IntelliJ IDEA 2024.2.0.1\plugins\junit\lib\junit5-rt.jar;E:\IntelliJ IDEA 2024.2.0.1\plugins\junit\lib\junit-rt.jar;E:\Hospital-Management-System\HospitalManagementSystemWebApp\target\test-classes;E:\Hospital-Management-System\HospitalManagementSystemWebApp\target\classes;C:\Users\User\.m2\repository\jakarta\servlet\jakarta.servlet-api\6.0.0\jakarta.servlet-api-6.0.0.jar;C:\Users\User\.m2\repository\org\junit\jupiter\junit-jupiter-api\5.10.0\junit-jupiter-api-5.10.0.jar;C:\Users\User\.m2\repository\org\opentest4j\opentest4j\1.3.0\opentest4j-1.3.0.jar;C:\Users\User\.m2\repository\org\junit\platform\junit-platform-commons\1.10.0\junit-platform-commons-1.10.0.jar;C:\Users\User\.m2\repository\org\apiguardian\apiguardian-api\1.1.2\apiguardian-api-1.1.2.jar;C:\Users\User\.m2\repository\org\junit\jupiter\junit-jupiter-engine\5.10.0\junit-jupiter-engine-5.10.0.jar;C:\Users\User\.m2\repository\org\junit\platform\junit-platform-engine\1.10.0\junit-platform-engine-1.10.0.jar;C:\Users\User\.m2\repository\com\mysql\mysql-connector-j\8.0.33\mysql-connector-j-8.0.33.jar;C:\Users\User\.m2\repository\com\google\protobuf\protobuf-java\3.21.9\protobuf-java-3.21.9.jar;C:\Users\User\.m2\repository\log4j\log4j\1.2.12\log4j-1.2.12.jar;C:\Users\User\.m2\repository\org\json\json\20240303\json-20240303.jar;C:\Users\User\.m2\repository\org\mockito\mockito-junit-jupiter\5.10.0\mockito-junit-jupiter-5.10.0.jar;C:\Users\User\.m2\repository\org\mockito\mockito-core\5.10.0\mockito-core-5.10.0.jar;C:\Users\User\.m2\repository\net\bytebuddy\byte-buddy\1.14.11\byte-buddy-1.14.11.jar;C:\Users\User\.m2\repository\net\bytebuddy\byte-buddy-agent\1.14.11\byte-buddy-agent-1.14.11.jar;C:\Users\User\.m2\repository\org\objenesis\objenesis\3.3\objenesis-3.3.jar" com.intellij.rt.junit.JUnitStarter -ideVersion5 -junit5 com.hospital.service.ServiceImplDiffblueTest
Process finished with exit code 0
</output>
    </root>
    <test duration="3264" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testGetAllPatient" name="testGetAllPatient()" metainfo="" status="passed">
        <output type="stderr">Java HotSpot(TM) 64-Bit Server VM warning: Sharing is only supported for boot loader classes because bootstrap classpath has been appended
</output>
    </test>
    <test duration="30" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testAddAuthUser2" name="testAddAuthUser2()" metainfo="" status="passed"/>
    <test duration="93" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testDeleteAuthUser" name="testDeleteAuthUser()" metainfo="" status="passed"/>
    <test duration="36" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testSearchPatient" name="testSearchPatient()" metainfo="" status="passed"/>
    <test duration="29" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testSearchUser" name="testSearchUser()" metainfo="" status="passed"/>
    <test duration="29" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testSearchUser2" name="testSearchUser2()" metainfo="" status="passed"/>
    <test duration="44" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testLogin" name="testLogin()" metainfo="" status="passed">
        <output type="stdout">[ERROR] main ServiceImpl : 560 - Error Occured at Service Layer: Invalid username or Password
[ERROR] main ServiceImpl : 561 - com.hospital.exception.DatabaseException: Invalid username or Password
</output>
    </test>
    <test duration="449" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testSearchAppointment" name="testSearchAppointment()" metainfo="" status="passed"/>
    <test duration="132" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testUpdateSchedule" name="testUpdateSchedule()" metainfo="" status="passed">
        <output type="stdout">[ERROR] main ServiceImpl : 310 - Schedule already exists for Date: 1970-01-01 and Time: 00:00 - 00:00
</output>
    </test>
    <test duration="193" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testViewAppointments2" name="testViewAppointments2()" metainfo="" status="passed"/>
    <test duration="39" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testSearchAppointment2" name="testSearchAppointment2()" metainfo="" status="passed"/>
    <test duration="89" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testSearchAppointment3" name="testSearchAppointment3()" metainfo="" status="passed">
        <output type="stdout">[ERROR] main ServiceImpl : 543 - Error Occured at Service Layer: Patient not present with id: 1
[ERROR] main ServiceImpl : 544 - com.hospital.exception.DatabaseException: Patient not present with id: 1
</output>
    </test>
    <test duration="18" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testGetAllPatient2" name="testGetAllPatient2()" metainfo="" status="passed"/>
    <test duration="26" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testAddPatient2" name="testAddPatient2()" metainfo="" status="passed"/>
    <test duration="44" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testAddPatient3" name="testAddPatient3()" metainfo="" status="passed">
        <output type="stdout">[ERROR] main ServiceImpl : 196 - Patient with id 1 already exists
</output>
    </test>
    <test duration="18" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testGetAllAuthUser2" name="testGetAllAuthUser2()" metainfo="" status="passed"/>
    <test duration="12" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testGetAllAuthUser3" name="testGetAllAuthUser3()" metainfo="" status="passed"/>
    <test duration="15" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testGetAllAuthUser" name="testGetAllAuthUser()" metainfo="" status="passed"/>
    <test duration="27" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testAddAuthUser" name="testAddAuthUser()" metainfo="" status="passed">
        <output type="stdout">[ERROR] main ServiceImpl : 74 - Username already exists
</output>
    </test>
    <test duration="121" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testViewAppointments" name="testViewAppointments()" metainfo="" status="passed"/>
    <test duration="38" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testBookAppointment" name="testBookAppointment()" metainfo="" status="passed"/>
    <test duration="193" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testCancelAppointment2" name="testCancelAppointment2()" metainfo="" status="passed"/>
    <test duration="19" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testCancelAppointment3" name="testCancelAppointment3()" metainfo="" status="passed">
        <output type="stdout">[ERROR] main ServiceImpl : 426 - Appointment doesnt exist with ID: 1
</output>
    </test>
    <test duration="75" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testCancelAppointment4" name="testCancelAppointment4()" metainfo="" status="passed"/>
    <test duration="136" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testCancelAppointment5" name="testCancelAppointment5()" metainfo="" status="passed"/>
    <test duration="21" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testCancelAppointment6" name="testCancelAppointment6()" metainfo="" status="passed">
        <output type="stdout">[ERROR] main ServiceImpl : 440 - Error Occured at Service Layer: An error occurred
[ERROR] main ServiceImpl : 441 - com.hospital.exception.DatabaseException: An error occurred
</output>
    </test>
    <test duration="123" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testCancelAppointment7" name="testCancelAppointment7()" metainfo="" status="passed"/>
    <test duration="35" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testCancelAppointment8" name="testCancelAppointment8()" metainfo="" status="passed">
        <output type="stdout">[ERROR] main ServiceImpl : 460 - Unauthoried access to cancel the Appointment: AppointmentTbl{AppointmentId=1, patient=PatientTbl{PatientId=1, name='String', email='String', mobileNo='String', addedByUser=AuthUserTbl{UserId=1, name='String', email='String', mobileNo='String', username='String', password='b2ef230e7f4f315a28cdcc863028da31f7110f3209feb76e76fed0f37b3d8580', role='String'}}, schedule=ScheduleTbl{ScheduleId=1, Doctor=AuthUserTbl{UserId=1, name='String', email='String', mobileNo='String', username='String', password='b2ef230e7f4f315a28cdcc863028da31f7110f3209feb76e76fed0f37b3d8580', role='String'}, scheduleDate=1970-01-01, startTime=00:00, endTime=00:00, scheduleStatus=1, unavailabilityReason='String'}, suggestedTests='String', suggestedMedicines='String', sufferingFromDisease='String', appointmentStatus=1} by Doctor: AuthUserTbl{UserId=1, name='null', email='null', mobileNo='null', username='janedoe', password='e4ad93ca07acb8d908a3aa41e920ea4f4ef4f26e7f86cf8291c5db289780a5ae', role='null'}
</output>
    </test>
    <test duration="27" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testCancelAppointment9" name="testCancelAppointment9()" metainfo="" status="passed">
        <output type="stdout">[ERROR] main ServiceImpl : 455 - Appointment doesnt exist with ID: 1
</output>
    </test>
    <test duration="17" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testSearchSchedule" name="testSearchSchedule()" metainfo="" status="passed"/>
    <test duration="25" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testBookAppointment2" name="testBookAppointment2()" metainfo="" status="passed"/>
    <test duration="29" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testBookAppointment3" name="testBookAppointment3()" metainfo="" status="passed">
        <output type="stdout">[ERROR] main ServiceImpl : 288 - Error Occured at Service Layer: Patient not found with ID: 1
[ERROR] main ServiceImpl : 289 - com.hospital.exception.DatabaseException: Patient not found with ID: 1
</output>
    </test>
    <test duration="10" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testAddPatient" name="testAddPatient()" metainfo="" status="passed"/>
    <test duration="10" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testAddSchedule2" name="testAddSchedule2()" metainfo="" status="passed">
        <output type="stdout">[ERROR] main ServiceImpl : 135 - Doctor with id 1 not found
</output>
    </test>
    <test duration="16" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testAddSchedule3" name="testAddSchedule3()" metainfo="" status="passed">
        <output type="stdout">[ERROR] main ServiceImpl : 145 - Schedule already exists
</output>
    </test>
    <test duration="19" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testAddSchedule4" name="testAddSchedule4()" metainfo="" status="passed">
        <output type="stdout">[ERROR] main ServiceImpl : 150 - You cannot add schedule for past date
</output>
    </test>
    <test duration="14" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testAddSchedule5" name="testAddSchedule5()" metainfo="" status="passed">
        <output type="stdout">[ERROR] main ServiceImpl : 155 - Schedule start time should be between 9AM to 6PM
</output>
    </test>
    <test duration="16" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testUpdateSchedule2" name="testUpdateSchedule2()" metainfo="" status="passed">
        <output type="stdout">[ERROR] main ServiceImpl : 305 - Doctor with id 1 not found
</output>
    </test>
    <test duration="11" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testUpdateSchedule3" name="testUpdateSchedule3()" metainfo="" status="passed">
        <output type="stdout">[ERROR] main ServiceImpl : 300 - Schedule with id 0 not found
</output>
    </test>
    <test duration="19" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testUpdateSchedule4" name="testUpdateSchedule4()" metainfo="" status="passed">
        <output type="stdout">[ERROR] main ServiceImpl : 315 - You cannot add schedule for past date
</output>
    </test>
    <test duration="16" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testUpdateAuthUser" name="testUpdateAuthUser()" metainfo="" status="passed"/>
    <test duration="5" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testNewServiceImpl" name="testNewServiceImpl()" metainfo="" status="passed"/>
    <test duration="25" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testCancelAppointment10" name="testCancelAppointment10()" metainfo="" status="passed">
        <output type="stdout">[ERROR] main ServiceImpl : 460 - Unauthoried access to cancel the Appointment: AppointmentTbl{AppointmentId=1, patient=PatientTbl{PatientId=1, name='String', email='String', mobileNo='String', addedByUser=AuthUserTbl{UserId=1, name='String', email='String', mobileNo='String', username='String', password='b2ef230e7f4f315a28cdcc863028da31f7110f3209feb76e76fed0f37b3d8580', role='String'}}, schedule=ScheduleTbl{ScheduleId=1, Doctor=AuthUserTbl{UserId=32, name='String', email='String', mobileNo='String', username='String', password='b2ef230e7f4f315a28cdcc863028da31f7110f3209feb76e76fed0f37b3d8580', role='String'}, scheduleDate=1970-01-01, startTime=00:00, endTime=00:00, scheduleStatus=1, unavailabilityReason='String'}, suggestedTests='String', suggestedMedicines='String', sufferingFromDisease='String', appointmentStatus=1} by Doctor: AuthUserTbl{UserId=1, name='null', email='null', mobileNo='null', username='janedoe', password='e4ad93ca07acb8d908a3aa41e920ea4f4ef4f26e7f86cf8291c5db289780a5ae', role='null'}
</output>
    </test>
    <test duration="21" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testCancelAppointment11" name="testCancelAppointment11()" metainfo="" status="passed">
        <output type="stdout">[ERROR] main ServiceImpl : 460 - Unauthoried access to cancel the Appointment: AppointmentTbl{AppointmentId=1, patient=PatientTbl{PatientId=1, name='String', email='String', mobileNo='String', addedByUser=AuthUserTbl{UserId=1, name='String', email='String', mobileNo='String', username='String', password='b2ef230e7f4f315a28cdcc863028da31f7110f3209feb76e76fed0f37b3d8580', role='String'}}, schedule=ScheduleTbl{ScheduleId=1, Doctor=AuthUserTbl{UserId=1, name='String', email='String', mobileNo='String', username='String', password='b2ef230e7f4f315a28cdcc863028da31f7110f3209feb76e76fed0f37b3d8580', role='String'}, scheduleDate=1970-01-01, startTime=00:00, endTime=00:00, scheduleStatus=1, unavailabilityReason='String'}, suggestedTests='String', suggestedMedicines='String', sufferingFromDisease='String', appointmentStatus=1} by Doctor: null
</output>
    </test>
    <test duration="10" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testAddSchedule" name="testAddSchedule()" metainfo="" status="passed">
        <output type="stdout">[ERROR] main ServiceImpl : 140 - User is not a doctor
</output>
    </test>
    <test duration="8" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testDeleteAuthUser2" name="testDeleteAuthUser2()" metainfo="" status="passed">
        <output type="stdout">[ERROR] main ServiceImpl : 112 - AuthUser with id 1 not found
</output>
    </test>
    <test duration="8" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testDeleteAuthUser3" name="testDeleteAuthUser3()" metainfo="" status="passed"/>
    <test duration="34" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testCancelAppointment" name="testCancelAppointment()" metainfo="" status="passed"/>
    <test duration="9" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testSearchSchedule2" name="testSearchSchedule2()" metainfo="" status="passed"/>
    <test duration="9" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testSearchPatient2" name="testSearchPatient2()" metainfo="" status="passed"/>
    <test duration="11" locationUrl="java:test://com.hospital.service.ServiceImplDiffblueTest/testSearchPatient3" name="testSearchPatient3()" metainfo="" status="passed">
        <output type="stdout">[ERROR] main ServiceImpl : 509 - Error Occured at Service Layer: User not present with id: 1
[ERROR] main ServiceImpl : 510 - com.hospital.exception.DatabaseException: User not present with id: 1
</output>
    </test>
</testrun>
